build-tools-internal/src/main/groovy/org/elasticsearch/gradle/internal/doc/SnippetsTask.groovy

```diff
@@ -67,6 +67,11 @@ class SnippetsTask extends DefaultTask {
          * the lines that make up the snippet.
          */
         for (File file: docs) {
+
+            if (! file.getName().endsWith(".asciidoc")) {
```

Comment:
I don't think this is the correct approach here, instead we ought to configure the task to ignore unwanted files in the first place. @mark-vieira can correct me, but I believe we need to edit `docs/build.gradle` and where we configure`docsFileTree`, we need something like

    include "**/*.asciidoc"

Reply:
Correct. This has the benefit of not having Gradle snapshot a bunch of unnecessary files as well as better build avoidance.